### PR TITLE
Extra Odbc Documentation

### PR DESCRIPTION
**Pull request by maiadegraaf**
This PR updates the ODBC README and explains the different handles, connections, error handling, binding, and buffers.  

It also includes a step-by-step explanation on how to create a basic ODBC application and a sample application, and the psqlodbc setup section is now replaced by the ODBC unit test info.

### CODE REVIEW

File: tools/odbc/README.md [hunk 1]
```diff
@@ -1,49 +1,306 @@
...
+| `SQL_NEED_DATA`         | More data is needed, such as when a parameter data is sent at execution time, or additional connection information is required.              |
+| `SQL_STILL_EXECUTING`   | A function that was asynchronously executed is still executing.                                                                              |
+
+### Buffers and Binding
+A buffer is a block of memory used to store data.  Buffers are used to store data retrieved from the database, or to send data to the database.  Buffers are allocated by the application, and then bound to a column in a result set, or a parameter in a query, using the [`SQLBindCol`](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlbindcol-function?view=sql-server-ver16) and [`SQLBindParameter`](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlbindparameter-function?view=sql-server-ver16) functions.  When the application fetches a row from the result set, or executes a query, the data is stored in the buffer.  When the application sends a query to the database, the data in the buffer is sent to the database.
+
+## Setting up an Application:
```
Comment (taniabogatsch) [inline tools/odbc/README.md:74]: Maybe we want to mention the OS that some parts of this tutorial are based on (e.g., `brew`)?

**Review by taniabogatsch — APPROVED** [review]
Very nice! I only have one minor comment.

Will we put this in another public place? What do you think @szarnyasg?

File: tools/odbc/README.md (hunk 1 above)
Comment (maiadegraaf) [reply tools/odbc/README.md:74]: Good point, I used the information that's already in the [docs](https://duckdb.org/docs/archive/0.8.1/api/odbc/overview) to install the driver, but should be included here as well!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**